src/se/vidstige/jadb/JadbConnection.java

```diff
@@ -38,9 +39,35 @@ public void getHostVersion() throws IOException, JadbException {
         devices.send("host:devices");
         devices.verifyResponse();
         String body = devices.readString();
+        devices.close();
         return parseDevices(body);
     }
 
+    public AsyncActionHandler watchDevices(final DeviceDetectionListener listener) throws IOException, JadbException {
+        final Transport transport = createTransport();
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    transport.send("host:track-devices");
+                    transport.verifyResponse();
+                    boolean r = false;
+                    do {
+                        List<JadbDevice> list = parseDevices(transport.readString());
+                        r = listener.detect(list);
+                    } while (r);
+                } catch (SocketException e) {
+                    // socket closed from another thread
+                } catch (Exception e) {
+                    Thread t = Thread.currentThread();
+                    t.getUncaughtExceptionHandler().uncaughtException(t, e);
+                }
+            }
+        }).start();
+
+        return new AsyncActionHandler(transport);
+    }
```

Comment:
I've been trying to avoid threads the greatest extend possible. I'm thinking a user of this library can create the threads themselves? So here we would just `parseDevices` and pass to listener like so

```java
public void watchDevices(final DeviceDetectionListener listener) throws IOException, JadbException {
    final Transport transport = createTransport();
    transport.send("host:track-devices");
    transport.verifyResponse();
   while (true) {
       List<JadbDevice> list = parseDevices(transport.readString());
       for (JadbDevice device : list) {
           listener.onDevice(device);
       }
   }
}
```

This way it will be easier to use in various contexts, etc.